src/planet-a/IcoSphereTree.cpp

```diff
@@ -159,10 +146,10 @@ void IcoSphereTree::initialize(float radius)
     for (int i = 0; i < gc_icosahedronFaceCount; i ++)
     {
         // Set triangles
-        SubTriangle tri;
+        SubTriangle &rTri = m_triangles.emplace_back();
 
         // indices were already calculated beforehand
-        set_verts(tri,
+        set_verts(rTri,
```

Comment:
My first impression is that this should instead be a constructor parameter.

My second impression is that it should be a member function.

Thoughts?

Reply:
New triangles are almost never created during runtime; existing triangle in the buffers are reused, so a constructor wouldn't be practical. A member function would be good.

Reply:
Actually, if std::array was used instead of c-style arrays, then `rTri.m_corners = {a, b, c}` can be done.